src/java_bytecode/java_entry_point.cpp

```diff
@@ -485,6 +485,49 @@ bool java_entry_point(
     pointer_type_selector);
 }
 
+/// Creates the initialize methods again taking account of symbols added to the
+/// symbol table during instantiation of lazy methods since they were first
+/// created,
+/// \param symbol_table: global symbol table containing symbols to initialize
+/// \param main_class: the class containing the "main" entry point
+/// \param message_handler: message_handlert for logging
+/// \param assume_init_pointers_not_null: specifies behaviour for
+/// java_static_lifetime_init
+/// \param object_factory_parameters: specifies behaviour for
+/// java_static_lifetime_init
+/// \param pointer_type_selector: specifies behaviour for
+/// java_static_lifetime_init
+bool recreate_initialize(
+  symbol_table_baset &symbol_table,
+  const irep_idt &main_class,
+  message_handlert &message_handler,
+  bool assume_init_pointers_not_null,
+  const object_factory_parameterst &object_factory_parameters,
+  const select_pointer_typet &pointer_type_selector)
+{
+  messaget message(message_handler);
+  main_function_resultt res=
+    get_main_symbol(symbol_table, main_class, message_handler);
+  if(res.status!=main_function_resultt::Success)
+  {
+    // No initialization was originally created (yikes!) so we can't recreate
+    // it now
+    return res.status==main_function_resultt::Error;
+  }
+  symbolt symbol=res.main_function;
```

Comment:
I'm not sure this is stable over `create_initialize`. If you can guarantee that it is then we could grab it at this point.

Reply:
I'd say those symbols should be stable, but better safe than sorry (as you'd be relying on implementation internals). So what I'd recommend is only taking a copy of the location rather than the full symbol, as this is all that you use.

Reply:
I'm worried that there may be a location numbering update when a new function is loaded but to be honest I don't really understand how these things work so maybe the worry is unfounded.

Reply:
In the final version of this series, line 502 is the only place where `symbol` is referred to, and then only `symbol.location` is used. Thus my suggestion to store that `source_locationt` rather than the full `symbolt`.

Reply:
`res` already contains a `symbolt` (i.e. a value type, not a reference), so I suspect it makes no difference whether we take an additional copy here or not

Reply:
Having looked at `create_initialize` it looks OK so I've moved the whole access later.